packages/synapse-react-client/src/utils/hooks/useLogin.test.tsx

```diff
@@ -403,21 +419,92 @@ describe('useLogin tests', () => {
       expect(result.current.step).toBe('VERIFICATION_CODE')
     })
 
+    // Change the step so the UI shows the correct prompt
+    act(() => {
+      result.current.onStepChange('DISABLE_2FA_PROMPT')
+    })
+    await waitFor(() => {
+      expect(result.current.step).toBe('DISABLE_2FA_PROMPT')
+    })
+
```

Comment:
Update the test to simulate what the components are doing. This alone doesn't catch the bug, because the screen flashes back after the `expect` completes.